File: datasets/IR-Plag/convertai-original-java-rest-python/case-06/plagiarized/L6/04/Kasus6.py
import sys

def simpan_array(arr, i):
    sc = sys.stdin
    if i > 0:
        simpan_array(arr, i - 1)
        print("Read a number: ", end="")
        arr[i - 1] = int(sc.readline().strip())

def tampil_array(arr, i):
    if i >= 0:
        print(arr[i])
        tampil_array(arr, i - 1)

def main():
    n = 10
    arr = [0] * n
    simpan_array(arr, n)
    tampil_array(arr, n - 1)

if __name__ == "__main__":
    main()

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-06/plagiarized/L5/04/Kasus6.ts
import * as readline from 'readline';

function simpanArray(arr: number[], n: number): void {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    for (let i = 0; i < n; i++) {
        rl.question("Read a number: ", (input) => {
            arr[i] = parseInt(input);
            if (i === n - 1) {
                rl.close();
            }
        });
    }
}

function tampilArray(arr: number[], i: number): void {
    console.log(arr[i]);
}

function main(): void {
    const n = 10;
    const arr: number[] = new Array(n);
    simpanArray(arr, n);
    for (let i = n - 1; i >= 0; i--) {
        tampilArray(arr, i);
    }
}

main();

